lib/game.rb

```diff
@@ -13,57 +15,56 @@ class Game
   def initialize
     @computer = Player.new('computer')
     @human = Player.new('human')
+    self.welcome
   end
 
-  # def welcome
-  #
-  #   puts "Welcome to BATTLESHIP"
-  #   puts "Enter p to play. Enter q to quit"
-  #   input = gets.chomp
-  #
-  #
-  #     if input == "p" || input == "P"
-  #       self.start_game
-  #     elsif input == 'q' || input == "Q"
-  #       exit
-  #     else
-  #       puts 'INVALID INPUT'
-  #       self.welcome
-  #     end
-  # end
+  def welcome
+    puts "Welcome to BATTLESHIP"
+    puts "Enter p to play. Enter q to quit"
+    input = gets.chomp
+
+      if input == "p" || input == "P"
+        self.start_game
+      elsif input == 'q' || input == "Q"
+        exit
+      else
+        puts 'INVALID INPUT'
+        self.welcome
+      end
+  end
 
   def start_game
     @computer = Player.new(:computer)
     @player = Player.new(:human)
-
-
   end
 
+
   def setup
     place_ships_ai
     place_ships_user
   end
 
-  def place_ships_user
+
+def place_ships_user
     puts "I have laid out my ships on the grid."
     puts "You now need to lay out your two ships."
     puts "The Cruiser is two units long and the Submarine is three units long."
     puts @human.board.render
 
     loop do
-    puts "Enter the squares for the Cruiser (3 spaces). Ex: (A1, A2, A3) \n"
-    input = gets.chomp.upcase
-    input_cruiser = input.split
+      puts "Enter the squares for the Cruiser (3 spaces). Ex: (A1 A2 A3) \n"
+      input = gets.chomp.upcase
+      input_cruiser = input.split
 
 
-    if @human.board.valid_placement?(@human.ships[:cruiser], input_cruiser) == true
-      @human.board.place(@human.ships[:cruiser], input_cruiser)
-      @human.board.render(true)
-      break
-    else
-      puts "Those are invalid coordinates. Please try again."
-    end
-  end #end loop
+      if @human.board.valid_placement?(@human.ships[:cruiser], input_cruiser) == true
```

Comment:
@m-mrcr we can delete the == true here, I don't think it's necessary.

Reply:
Yeah, I agree.